catalog/src/main/java/org/killbill/billing/catalog/StandaloneCatalogWithPriceOverride.java

```diff
@@ -53,16 +55,16 @@
     public StandaloneCatalogWithPriceOverride() {
     }
 
-    public StandaloneCatalogWithPriceOverride(final StandaloneCatalog catalog, final PriceOverride priceOverride, final Long tenantRecordId, final InternalCallContextFactory internalCallContextFactory) {
+    public StandaloneCatalogWithPriceOverride(final StaticCatalog catalog, final PriceOverride priceOverride, final Long tenantRecordId, final InternalCallContextFactory internalCallContextFactory) {
         // Initialize from input catalog
         setCatalogName(catalog.getCatalogName());
         setEffectiveDate(catalog.getEffectiveDate());
-        setProducts(catalog.getCurrentProducts());
-        setPlans(catalog.getCurrentPlans());
-        setPriceLists(catalog.getPriceLists());
-        setPlanRules(catalog.getPlanRules());
-        setSupportedCurrencies(catalog.getCurrentSupportedCurrencies());
-        setUnits(catalog.getCurrentUnits());
+        setProducts(((StandaloneCatalog) catalog).getCurrentProducts());
```

Comment:
The cast is only to handle the exception? Would it be easier instead to propagate the exception to the constructor?

Reply:
I have a better idea: `https://github.com/killbill/killbill-api/pull/66/commits/5d8f1a7d5943bf058227ecc7ae56f702f4035a13` (the reason for such exceptions was that both `DefaultVersionedCatalog` was implementing both `Catalog` and `StaticCatalog` and exceptions were needed when we could not find the right catalog version.


I will then look to remove casts related to this.